sharktank/tests/models/llama4/llama4_test_iree_eager.py

```diff
@@ -0,0 +1,411 @@
+
+    if self.expert_shared_count is not None:
+        moe_output = moe_output + self.shared_experts(ffn_input)
+
+    moe_output = moe_output.reshape(batch_size, sequence_length, feature_dim)
+
+    moe_output = self.layer_output_norm(moe_output)
+
+    return moe_output
+
+
+def export_llama4_model_mlir(
+    model_or_parameters_path: PagedLlmModelV1 | PathLike,
+    /,
+    output_path: PathLike,
+    batch_size: int,
+    config: LlamaModelConfig,  # TODO; it can be read from HF when comparing the actual model
+):
+    mlir_path = output_path / "model.mlir"
+    export_config_path = output_path / "model_export_config.json"
+    irpa_path = output_path / "parameters.irpa"
+    export_artifacts = ExportArtifacts.from_config(
+        config,
+        irpa_path=irpa_path,
+        batch_size=batch_size,
+        iree_hip_target="gfx942",
+        iree_hal_target_device="hip",
+        # iree_hal_local_target_device_backends=self.iree_hal_local_target_device_backends,
+        use_attention_mask=True,
+    )
+
+    cli_args = [
+        "export_paged_llm_v1",
+        "--irpa-file",
+        str(irpa_path),
+        "--output-mlir",
+        str(mlir_path),
+        "--output-config",
+        str(export_config_path),
+        "--bs-prefill",
+        str(batch_size),
+        "--bs-decode",
+        str(batch_size),
+        "--block-seq-stride",
+        str(config.block_seq_stride),
+        "--attention-dtype",
+        "float16",
+        "--activation-dtype",
+        "float16",
+        "--tensor-parallelism-size",
+        "1",
+        "--pipeline-parallelism-size",
+        "1",
+        "--attention-kernel",
+        "torch",
+        "--skip-decode",
+        "--use-attention-mask",
+    ]
+
+    sys_argv_backup = sys.argv
+    sys.argv = cli_args
+    export_paged_llm_v1.main()
+
+    sys.argv = sys_argv_backup
+
+    iree_module_path = output_path / "model.vmfb"
+    export_artifacts.compile_to_vmfb(
+        output_mlir=mlir_path,
+        output_vmfb=iree_module_path,
+        args=[],
+    )
+
+
+def export_llama4_iree_parameters(
+    model: PagedLlmModelV1,
+    parameters_output_path: PathLike,
+    config: LlamaModelConfig,  # TODO; it can be read from HF when comparing the actual model
+):
+    config_dict = {
+        "hparams": asdict(config.hp),
+    }
+
+    dataset = Dataset(config_dict, root_theta=model.theta)
+    dataset.save(parameters_output_path)
+
+
+def export_llama4(
+    model: PagedLlmModelV1,
+    mlir_output_path: PathLike,
+    parameters_output_path: PathLike,
+    output_path: PathLike,
+    batch_size: int,
+    config: LlamaModelConfig,  # TODO; it can be read from HF when comparing the actual model
+):
+    export_llama4_iree_parameters(model, parameters_output_path, config=config)
+    export_llama4_model_mlir(
+        model, output_path=output_path, config=config, batch_size=batch_size
+    )
+
+
+def runCompareIreeAgainstTorchEager(args, atol):
```

Comment:
Need to rewrite this so that it gets picked up by pytest and automatically run.

Also likely needs a `@is_mi300x` decorator.